head.go

```diff
@@ -640,13 +640,20 @@ func (h *Head) Appender() Appender {
 func (h *Head) appender() *headAppender {
 	return &headAppender{
 		head:         h,
-		minValidTime: h.MaxTime() - h.chunkRange/2,
+		minValidTime: max(h.MinTime(), h.MaxTime()-h.chunkRange/2),
```

Comment:
I am not entirely sure about this one so let me know if you think it might brake anything else.

Reply:
> adding samples with timestamps lower than the first appended sample.

That's a valid use case.